File: 20 Valid Parentheses/soln.go
func isValid(s string) bool {
    var stack []string
    
    for i := 0; i < len(s); i++{
        l := string(s[i])
        if l == "(" || l == "[" || l == "{"{
            stack = append(stack, l)
            continue;
        } 
        if len(stack) == 0{
            return false;
        }
        p := stack[len(stack) - 1]
        stack = stack[:len(stack) - 1]
        
        if l == ")" && p != "("{
            return false
        }
        if l == "]" && p != "["{
            return false
        }
        if l == "}" && p != "{"{
            return false
        }
    }
    return len(stack) == 0
}

File: 20 Valid Parentheses/soln.cpp
class Solution {
public:
    bool isValid(string s) {
        vector<char> Stack;
        
        for (int i = 0; i < s.size(); i++){
            string l;
            l = s[i];
            if (l == "(" || l == "{" || l == "["){
                Stack.push_back(s[i]);
                continue;
            }
            if (Stack.size() == 0){
                return false;
            }
            string p;
            p = Stack[Stack.size() - 1];
            Stack.pop_back();
            
            if (l == ")" && p != "("){
                return false;
            }
            if (l == "]" && p != "["){
                return false;
            }
            if (l == "}" && p != "{"){
                return false;
            }
        }
        return Stack.size() == 0;
    }
};